helpdesk/views/staff.py

```diff
@@ -422,8 +423,13 @@ def view_ticket(request, ticket_id):
 
         return redirect('helpdesk:edit_ticket_checklist', ticket.id, checklist.id)
 
+    # Open tickets on top
+    dependencies = list(ticket.ticketdependency.filter(depends_on__status__in=Ticket.OPEN_STATUSES)) \
+        + list(ticket.ticketdependency.filter(depends_on__status__not_in=Ticket.OPEN_STATUSES))
```

Comment:
`not_in` lookup is more readable so no need to change it, but I think `ticket.ticketdependency.exclude(depends_on__status__in=Ticket.OPEN_STATUSES)` would do the same

Reply:
You are right. If you followed the related discussion you might see how we got here. The primary goal - a 'fast' way to concatentate two QuerySets in order - cannot be achieved anyway.  So I might as well remove the NotIn ORM operator, since this is the only place where it is used to remove code complexity.

There might be a way to do the chaining via annotations, as show [here](https://stackoverflow.com/questions/17216036/how-to-combine-two-select-queries-with-their-own-order) for SQL.

Reply:
Yep, this could be a readable solution! Too bad there isn't an easy one for that king of thing

Reply:
Please do not use EXCLUDE for this case.
It produces a very different set of underlying SQL that will be come a performance issue for deployments with large numbers of tickets.

Reply:
```
from django.db.models import Value
open_deps = ticket.ticketdependency.filter(depends_on__status__in=Ticket.OPEN_STATUSES)).annotate(rank=Value(1))
not_open_deps = ticket.ticketdependency.filter(depends_on__status__not_in=Ticket.OPEN_STATUSES)).annotate(rank=Value(2))
dependencies=open_deps.union(not_open_deps).order_by('rank')
```

Better?

@uhurusurfa : Please elaborate or point to some information regarding `.exclude` performance penalty.
A quick lookup in the web did not bring up substantial related information for me.

Reply:
I think I have found now an optimal solution using conditional annotation, please review the last commit.

Reply:
Yes this looks better!